src/uu/tr/src/operation.rs

```diff
@@ -672,18 +685,28 @@ where
     R: BufRead,
     W: Write,
 {
-    let mut buf = [0; 8192];
-    let mut output_buf = Vec::new();
+    let mut buf = [0; 32768]; // Large buffer for better throughput
+    let mut output_buf = Vec::with_capacity(32768); // Pre-allocated output buffer
+
+    loop {
+        let length = match input.read(&mut buf[..]) {
+            Ok(0) => break, // EOF reached
+            Ok(len) => len,
+            Err(e) if e.kind() == std::io::ErrorKind::Interrupted => continue,
+            Err(e) => return Err(e.map_err_context(|| translate!("tr-error-read-error"))),
+        };
 
-    while let Ok(length) = input.read(&mut buf[..]) {
-        if length == 0 {
-            break; // EOF reached
+        // Process the buffer in place, only adding translated chars to output
+        output_buf.clear();
+        for &byte in &buf[..length] {
+            if let Some(translated) = translator.translate(byte) {
+                output_buf.push(translated);
+            }
         }
 
-        let filtered = buf[..length]
-            .iter()
-            .filter_map(|&c| translator.translate(c));
-        output_buf.extend(filtered);
+        if output_buf.is_empty() {
```

Comment:
This looks like a micro-optimization for a rare case. I assume write_all does the same test internally. Maybe remove?